### ISSUE TITLE
Deauth: Terminate Program if user manually locks system

### ISSUE THREAD

**Issue by zakuArbor** [zakuArbor/proxyAuth#32]
**tldr:** Kill `deauth` program when the user locks their computer manually.

**tldr Approach:** Listen to the changes in state of `status` in session dbus the interface: `org.gnome.SessionManager.Presence` 

Currently, `deauth` can only be terminated via three ways:
1. Kill the process
2. Failed to Authenticate (In our current implementation, it's just failure to respond on time)
3. Log off the machine which effectively kills all processes in the session

**Issue:**
An issue can arise where a user logs in via `proxy_auth` PAM. A `deauth` program will be executed but what if the user chooses to manually lock their system? There will be multiple instances of `deauth` that can be created. The reason why the `deauth` program may not be killed is because the user simply locked their computer to take a "break" and be within the vicinity of the system and therefore still able to respond to the challenge and response of the system. However, when they wish to log back into their session, the user just needs to be in the login screen (not the general lock screen) which will spawn another `deauth`. This can cause a conflict because there'll be two bluetooth servers advertising the same uuid which is a big issue. From what I observed, it causes an infinite loop where one server connects to the laptop itself somehow. Not sure how this occurs exactly but I do know it's an issue.

**Potential Solution:**
Looking into various documentations, source code from various desktop environment vendors including GNOME and PopOS, and StackOverFlow, I think the best approach is to check the value of `status` from `org.gnome.SessionManager.Presence`

According to the [documentation](https://github.com/gcampax/gnome-session/blob/master/gnome-session/org.gnome.SessionManager.Presence.xml):
```
<property name="status" type="u" access="readwrite">
      <doc:doc>
        <doc:description>
          <doc:para>
            The status of the session.
          </doc:para>
          <doc:para>
            The status parameter must be one of the following:
            <doc:list>
              <doc:item>
                <doc:term>0</doc:term>
                <doc:definition>Available</doc:definition>
              </doc:item>
...
              <doc:item>
                <doc:term>3</doc:term>
                <doc:definition>Idle</doc:definition>
```
We can just listen to changes on the property `status` if it ever becomes a 3. We will be monitoring (i.e. adding the change of `status` property to our watch list). 

I see that GNOME has a good example of how to add things to our `watch` list: https://gitlab.gnome.org/GNOME/glib/-/blob/master/gio/tests/gdbus-example-watch-proxy.c

I recall seeing this a month and a few weeks ago but I thought the watch literally meant a watch but now I am slowly learning more about DBus, I realized I was completely wrong.

**Comment by zakuArbor** [comment]
**dbus type:** `G_BUS_TYPE_SESSION`
**dbus name:**  `"org.gnome.SessionManager"` 
**dbus obj path:** `"/org/gnome/SessionManager/Presence"` 
**dbus interface** `"org.gnome.SessionManager.Presence"`
